### PR TITLE
8292566: Add reference to the java.nio.file package in java.nio package documentation

### PR DESCRIPTION
**Pull request by bplb**
Add a phrase about `java.nio.file` to the sentence in the `java.nio` package documentation describing its subpackages.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change requires a CSR request to be approved

### Issues
 * [JDK-8292566](https://bugs.openjdk.org/browse/JDK-8292566): Add reference to the java.nio.file package in java.nio package documentation
 * [JDK-8292567](https://bugs.openjdk.org/browse/JDK-8292567): Add reference to the java.nio.file package in java.nio package documentation (**CSR**)

### Reviewers
 * [Iris Clark](https://openjdk.org/census#iris) (@irisclark - **Reviewer**) ⚠️ Review applies to [b7279eb9](https://git.openjdk.org/jdk/pull/9910/files/b7279eb926bfecfbfff9c63d2a1189626f5089d8)
 * [Jaikiran Pai](https://openjdk.org/census#jpai) (@jaikiran - **Reviewer**) ⚠️ Review applies to [b7279eb9](https://git.openjdk.org/jdk/pull/9910/files/b7279eb926bfecfbfff9c63d2a1189626f5089d8)
 * [Alan Bateman](https://openjdk.org/census#alanb) (@AlanBateman - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/9910/head:pull/9910` \
`$ git checkout pull/9910`

Update a local copy of the PR: \
`$ git checkout pull/9910` \
`$ git pull https://git.openjdk.org/jdk pull/9910/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 9910`

View PR using the GUI difftool: \
`$ git pr show -t 9910`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/9910.diff">https://git.openjdk.org/jdk/pull/9910.diff</a>

</details>

### CODE REVIEW

**Review by irisclark — APPROVED** [review]
Associated CSR also Reviewed.

**Review by jaikiran — APPROVED** [review]
Hello Brian, the change looks fine to me. The copyright year on this file will need an update.

**Comment by jaikiran** [comment]
After reading the text again, I had a question which I added to the CSR issue.

File: src/java.base/share/classes/java/nio/package-info.java [hunk 1]
```diff
@@ -54,9 +54,10 @@
  *
  * <p> The {@code java.nio} package defines the buffer classes, which
  * are used throughout the NIO APIs.  The charset API is defined in
- * the {@link java.nio.charset} package, and the channel and selector
- * APIs are defined in the {@link java.nio.channels} package.  Each of
- * these subpackages has its own service-provider (SPI) subpackage,
+ * the {@link java.nio.charset} package, the channel and selector APIs
+ * in the {@link java.nio.channels} package, and the file, file attribute,
+ * and file system APIs in the {@link java.nio.file} package.  Each of
+ * these subpackages has its own service-provider interface (SPI) subpackage,
```
[the remark this answers is not in the document]
Comment (bplb) [reply src/java.base/share/classes/java/nio/package-info.java]: Done in 53cdc0869aca7237a8703b05c1dd8f3c3db98796.

**Comment by bplb** [comment]
> The copyright year on this file will need an update.

Year updated in 53cdc0869aca7237a8703b05c1dd8f3c3db98796.

**Comment by bplb** [comment]
> [...] I had a question which I added to the CSR issue.

Mostly addressed in 53cdc0869aca7237a8703b05c1dd8f3c3db98796.

**Comment by bplb** [comment]
Note: I am not updating the CSR until there is consensus in this PR.

File: src/java.base/share/classes/java/nio/package-info.java [hunk 2]
```diff
@@ -50,13 +50,31 @@
...
+ *
+ *   <li><p> <a
+ *   href="file/Path.html"><i>Paths</i></a>,
+ *   which represent a hierarchical sequence of <br> name elements
+ *   used to locate a file in a file system.  </p></li>
+ *
+ *   <li><p> <a
+ *   href="file/attribute/package-summary.html"><i>Attributes</i></a>,
+ *   which represent <i>metadata</i> associated with <br> objects in a
+ *   file system.  </p></li>
```
Comment (AlanBateman) [inline src/java.base/share/classes/java/nio/package-info.java:67]: What would you think about reducing this down to "Path, which together with the Files class provide access to files.". The reason is that "FileSystem" isn't something that most developers will use directly and I don't think the attributes package is listed later and I don't think needs to be in the introducing to the abstractions.

Comment (bplb) [reply src/java.base/share/classes/java/nio/package-info.java]: I agree that the list was too long. Fixed in e72437776a0625f526934294d2df4cab1d848056.

**Review by AlanBateman — APPROVED** [review]
Update looks good.

**Pull request closed without merging**